### PR TITLE
refactor: Use ESBuild in prod if a user does not require Babel

### PR DESCRIPTION
**Pull request by rschristian**
Called out by @ArnaudBarre a few months ago on the Vite Discord (if memory serves), cheers to him for spotting this

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -160,12 +165,15 @@ function preactPlugin({
 			config = resolvedConfig;
 			devToolsEnabled =
 				devToolsEnabled ?? (!config.isProduction || devtoolsInProd);
+
+			useBabel =
+				!config.isProduction || devToolsEnabled || typeof babel !== "undefined";
```
Comment (rschristian) [inline src/index.ts:170]:
```suggestion
			useBabel =
				!config.isProduction || devToolsEnabled || typeof babel !== "undefined";
```

File: src/index.ts [hunk 2]
```diff
@@ -151,6 +152,10 @@ function preactPlugin({
 						},
 					},
 				},
+				esbuild: {
+					jsx: "automatic",
+					jsxImportSource: jsxImportSource ?? "preact",
```
Comment (marvinhagemeister) [inline src/index.ts:157]: IIrc there is a config option to enable the jsx dev transform in esbuild too.

Comment (rschristian) [reply src/index.ts:157]: Yeah could do, but that'd really only come up if the user has explicitly disabled devtools, which I'm (anecdotally) guessing is uncommon?

**Review by marvinhagemeister — APPROVED** [review]

File: src/index.ts (hunk 2 above)
Comment (rschristian) [reply src/index.ts:157]: Ah yeah, it's a bit unintuitive, but it's a Babel or ESBuild situation, not both.

Will add a comment to point that out.

File: src/index.ts [hunk 3]
```diff
@@ -151,6 +152,10 @@ function preactPlugin({
 						},
 					},
 				},
+				esbuild: {
```
Comment (rschristian) [inline src/index.ts:155]:
```suggestion
				// While this config is unconditional, it'll only be used if Babel is not
				esbuild: {
```

**Pull request merged**